docs/operation/troubleshooting.rst

```diff
@@ -0,0 +1,124 @@
+
+Errors after restarting the Bootstrap node
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you restart the Bootstrap node and for some reason, some containers
+(especially the salt-master container) refuses to start, check and ensure that
+the **MetalK8s ISO** is mounted properly.
+
+Bootstrap fails and console log is unscrollable
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If during a MetalK8s installation, the Bootstrap process fails and the console
+output is unscrollable then you can consult the Bootstrap logs in
+``/var/log/metalk8s-bootstrap.log``.
+
+Account Administration Errors
++++++++++++++++++++++++++++++
+
+Forgot the MetalK8s GUI password
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you forget the MetalK8s GUI username and/or password combination,
+follow
+:ref:`this procedure <Administering-MetalK8s-GUI-Kubernetes-API-and-Salt-API>`
+to reset or change it.
+
+General Kubernetes Resource Errors
+++++++++++++++++++++++++++++++++++
+
+Pod status shows "CrashLoopBackOff"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If after a MetalK8s installation, you notice some Pods are in a state of
+"CrashLoopBackOff", then it means pods are crashing because they start up then
+immediately exit, thus Kubernetes restarts them and the cycle continues.
+To get possible clues about this error, run the following commands and inspect
+the output.
+
+.. code-block:: shell
+
+   kubectl -n <namespace-name> describe pod <pod name>
+
+   kubectl -n <namespace-name> logs -p  <pod name>
+
+Persistent Volume Claim(PVC) stuck in "Pending" state
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If after provisioning a volume for a Pod say Prometheus and the PVC still
+hangs in a ``Pending`` state, then try checking the following.
+
+Check that the volumes have been provisioned and it's status is ``Ready``:
+
+.. code-block:: shell
+
+   kubectl describe volume <volume name>
+
+Check that a corresponding Persistent volume exist:
+
+.. code-block:: shell
+
+   kubectl get pv
+
+If no Persistent volume exist, then check that the storage operator is up
+and running.
+
+.. code-block:: shell
+
+   kubectl -n kube-system get pods
+
+Access to MetalK8s GUI fails with "undefined backend"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If in the cause of using the MetalK8s GUI, you encounter an "undefined
+backend" error then perform the following checks.
+
+Check that the Ingress pods are running:
+
+.. code-block:: shell
+
+   kubectl -n metalk8s-ingress get pods
+
+Check the Ingress controller logs:
+
+.. code-block:: shell
+
+   kubectl -n metalk8s-ingress logs <nginx-ingress-pod-name>
+
+Pod and Service CIDR conflicts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If after installation of a MetalK8s cluster you notice that Pod to pod
+communication has routing problems, perform the following.
+
+Check that Pod network subnets do not conflict with host networks by verifying
+allocated pod IP addresses:
```

Comment:
Maybe not the most straightforward approach. Let's add a first check on the actual network values:

```rst
- Check the configured values for the internal Pod and Service networks:

   .. code-block:: shell

      root@bootstrap # salt-call pillar.get metalk8s:networks

   Make sure the configured IP ranges (CIDR notation) do not conflict with your infrastructure.
```

After a check like this one, there shouldn't be a need to go through the `kubectl get pods` command. Going further may look into more advanced commands, such as `traceroute` or `tcpdump`. Not sure it's worth going too far in this iteration though.